src/libs/freetype/lv_freetype_outline.c

```diff
@@ -331,10 +331,48 @@ static lv_freetype_outline_t outline_create(
         }
     }
 
+    FT_Outline_Funcs outline_funcs = {
+        .move_to = outline_move_to_cb,
+        .line_to = outline_line_to_cb,
+        .conic_to = outline_conic_to_cb,
+        .cubic_to = outline_cubic_to_cb,
+        .shift = 0,
+        .delta = 0
+    };
+
+
     lv_result_t res;
     lv_freetype_outline_event_param_t param;
-
     lv_memzero(&param, sizeof(param));
+
+    /*Calculate Total Segmenets Before decompose */
+    int tag_size = face->glyph->outline.n_points;
+    int segments = 0;
+    int vectors = 0;
+
+    for(int j = 0; j < tag_size; j++) {
+        if((face->glyph->outline.tags[j] & 0x1) == 0x1) {
+            segments++;
+            vectors++;
+        }
+        else {
+            int jj = j + 1 < tag_size ? j + 1 : 0;
+            if(face->glyph->outline.tags[jj] & 0x1) {
+                vectors++;
+            }
+            else {
+                segments++;
+                vectors += 2;
+            }
+        }
+    }
+    /*Also for every contour we may have a line for close*/
+    segments += face->glyph->outline.n_contours;
+    vectors += face->glyph->outline.n_contours;
+
+    param.sizes.data_size = vectors * 2;
```

Comment:
Sorry, just for my understanding, why is it `* 2`?

Reply:
We count the number of floats, so it is vectors * 2